hub/policy/src/main/java/uk/gov/ida/hub/policy/domain/controller/IdpSelector.java

```diff
@@ -54,6 +64,25 @@ public static IdpSelectedState buildIdpSelectedState(IdpSelectingState state,
         );
     }
 
+    private static List<LevelOfAssurance> getLevelsOfAssuranceForTransactionSupportedByIdp(boolean idpEnabledForRegistration,
+                                                                                           List<LevelOfAssurance> idpLevelsOfAssurance,
+                                                                                           List<LevelOfAssurance> levelsOfAssuranceForTransaction) {
+        List<LevelOfAssurance> levelsOfAssuranceForTransactionSupportedByIdp;
+
+        if (!idpEnabledForRegistration
```

Comment:
I guess this is written for scenarios where there are LOAs that aren't just 1 or 2 but, as that's all we have right now and on the basis of YAGNI, you could (unless I'm mistaken which is likely enough) simplify this to:

`if(!idpEnabledForRegistration && levelsOfAssuranceForTransaction.size() == 2) {
    levelsOfAssuranceForTransaction = List.of(LevelOfAssurance.LEVEL_1, LevelOfAssurance.LEVEL_2)
}`

and maybe even (although I don't think simplify is necessarily the correct term anymore):

`if(!idpEnabledForRegistration) {
    Collections.sort(levelsOfAssuranceForTransaction) // I'm assuming without checking the sort is ascending by default.
}`

Reply:
That makes a lot more sense, John. Good suggestion. I'm thinking to use your first suggestions as it is more explicit.

Reply:
I guess as it is now makes it clearer in stating the scenario in which the LOAs need to be switched? I'm thinking what would make more sense to someone reading this down the line 🤷‍♂️ But setting it using List.of is better so I'm going to change that and leave the rest for more feedback.

Reply:
I've had to update this based on newly discovered requirements. So in the case where registration is not enabled for the service and they have the LOA mapping of [LOA2, LOA1] then we now only want to send them LOA2.